hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -216,6 +233,9 @@ protected List<HoodieFileGroup> buildFileGroups(Stream<HoodieBaseFile> baseFileS
           group.addNewFileSliceAtInstant(pendingCompaction.get().getKey());
         }
       }
+      if (logFiles.containsKey(pair)) {
+        logFiles.get(pair).stream().sorted(HoodieLogFile.getLogFileComparator()).forEach(logFile -> group.addLogFile(completionTimeQueryView, logFile));
+      }
```

Comment:
ok i see, here it's being sorted so log files will be added in sequence of deltacommit time. As long as there is one caller it's fine. But, how do we that other callers don't abuse it? Should we then add a validation in filegroup.addLogFile?

Reply:
Yeah, a validation maybe good, personally I just don't want too much over-protection logic.

Reply:
But not being in sorted order could lead to incorrect results right?

Reply:
Yes, but the only caller already guarantees the sorting order, still think adding a validation is not necessary.